prow/jenkins/jenkins.go

```diff
@@ -388,6 +388,27 @@ func (c *Client) doRequest(method, path string) (*http.Response, error) {
 	return c.client.Do(req)
 }
 
+// getJobName generates the correct job name for this job type
+func getJobName(spec *prowapi.ProwJobSpec) string {
+	if spec.JenkinsSpec != nil && spec.JenkinsSpec.GitHubBranchSourceJob && spec.Refs != nil {
```

Comment:
For GH-based jobs, they will always be associated with code but normally they're triggered by a GH event. So, maybe it doesn't make sense for periodics to have it... 🤔 Because I think all the refs and such would have to be manually put in the config? Maybe, unless there's a need, only pre/post could have it.

I wasn't sure if `Refs` are _always_ there (maybe something else generates jobs 🤷‍♂️ so i figured better check them). We don't  have to test `spec.JenkinsSpec.GitHubBranchSourceJob`, tho.

Reply:
> Because I think all the refs and such would have to be manually put in the config?

Is this referring to `ExtraRefs` in the utility config? https://godoc.org/k8s.io/test-infra/prow/config#UtilityConfig There is no way to manually supply a ProwJob's `spec.Refs` field from config. That field is populated automatically for presubmits and postsubmits based on the GH event that triggered them.

> I wasn't sure if Refs are always there 

`Refs` is never there for periodics and is always there for presubmit and postsubmit jobs.

> We don't have to test `spec.JenkinsSpec.GitHubBranchSourceJob`, tho

I don't follow. If we don't use `GitHubBranchSourceJob` here then it is only used for log message fields meaning it would have no functional impact.

Reply:
Sorry, nevermind the `GitHubBranchSourceJob` comment.

Ok, so re-reading the doc – it sounds like periodics don't need to support this. I have not personally used a periodic job but that's what it seems like.